pkg/logcli/output/output.go

```diff
@@ -0,0 +1,44 @@
+package output
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+// LogOutput is the interface any output mode must implement
+type LogOutput interface {
+	Format(ts time.Time, lbls *labels.Labels, maxLabelsLen int, line string) string
+	Print(ts time.Time, lbls *labels.Labels, maxLabelsLen int, line string)
```

Comment:
I would remove this function from the `interface`. It is kind of redundant to make every implementation include `fmt.Println(line)`. This could be easily handled outside, either by just returning a string to logcli and printing it there, or by wrapping the actual output:

```go
type LogFormatter interface {
  Format(ts time.Time, lbls *labels.Labels, maxLabelsLen int, line string) string
}

type LogOutput struct {
  f *LogFormatter
}

func (o LogOutput) Print(ts time.Time, lbls *labels.Labels, maxLabelsLen int, line string) {
  fmt.Println(o.f.Format(ts, lbls, maxLabelsLen, line)
}

func NewLogOutput(mode string, options *LogOutputOptions) (LogOutput, error) {
  var formatter *LogFormatter  
  switch mode {
    // ...
  }
  return LogOutput{f: formatter}
}

Reply:
I agree with you. I'm going to pick the shortest path here (calling `fmt.Println()` from outside) and we can eventually re-iterate on it on a second step, if you agree.